src/include/optimizer/remove_unnecessary_join_optimizer.h

```diff
@@ -6,16 +6,17 @@
 namespace kuzu {
 namespace optimizer {
 
-// Due to the nature of graph pattern, a (node)-[rel]-(node) is always interpreted as two joins.
-// However, in many cases, a single join is sufficient.
-// E.g. MATCH (a)-[e]->(b) RETURN e.date
-// Our planner will generate a plan where the HJ is redundant.
-//      HJ
-//     /  \
-//   E(e) S(b)
-//    |
-//   S(a)
-// This optimizer prunes such redundant joins.
+/* Due to the nature of graph pattern, a (node)-[rel]-(node) is always interpreted as two joins.
```

Comment:
Interesting, was this causing a warning?

Reply:
It was giving the warning `multi-line comment [-Werror=comment` specifically on the line with the backslash

From the gcc docs:
> Warn whenever a comment-start sequence ‘/*’ appears in a ‘/*’ comment, or whenever a backslash-newline appears in a ‘//’ comment. This warning is enabled by -Wall. 

Basically it thinks you're trying to use a single line comment as a multi-line by including a backslash+newline:
```cpp
//      HJ
//     /  \
     E(e) S(b)
//    |
//   S(a)
```
Because technically the above would be valid.